File: src/main/scala/com/vc/easy/L422.scala
package com.vc.easy

object L422 {
  def validWordSquare(words: List[String]): Boolean = {
    import scala.collection.mutable
    words.indices.foreach(i => {
      val h = words(i)
      val v = new mutable.StringBuilder()
      words.indices.foreach(j => {
        if(i < words(j).length) v.append(words(j)(i))
      })
      if(h != v.toString()) return false
    })
    true
  }
}

File: src/main/java/com/vc/easy/L422.java
package com.vc.easy;

import java.util.*;

class L422 {
    public boolean validWordSquare(List<String> words) {
        for(int h = 0; h < words.size(); h++) {
            String horizontalString = words.get(h);
            String verticalString = "";
            for(int v = 0; v < words.get(h).length(); v++) {
                verticalString += words.get(v).charAt(h);
            }
            if(!horizontalString.equals(verticalString)) return false;
        }
        return true;
    }
}
